src/storage/src/render/sources.rs

```diff
@@ -79,19 +80,56 @@ where
         })
         .as_collection();
 
-    ImportedTable {
+    RenderedTable {
         ok_collection,
         err_collection,
         handle,
         capability,
     }
 }
 
-/// Constructs a `CollectionBundle` and tokens from source arguments.
+/// Constructs a data [`Collection`], an error [`Collection`],
+/// and tokens from a [`SourceInstanceDesc`].
 ///
 /// The first returned pair are the row and error collections, and the
 /// second is a token that will keep the source alive as long as it is held.
-pub(crate) fn import_source<G>(
+///
+/// `render_source` _renders_ complete _differential_ streams that represent
+/// the final source as requested by the original `CREATE SOURCE` statement.
+/// This request is encapsulated in the passed [`SourceInstanceDesc`].
+///
+/// Note that this function attaches a single _instance_ of a source
+/// to the passed [`Scope`], and this instance may be disabled if the
+/// source-type does not support multiple-instances. `render_source`
+/// is called on each timely worker as part of
+/// [`super::build_storage_dataflow`].
+///
+/// The process of rendering is as follows:
+/// - If the requested source is a table ("`Local`"), then call [`render_table`].
+/// - Otherwise:
+///   - Build the _base_ source (the stream of data and/or errors coming from the upstream
+///   service:
+///     - If the requested source is a `SimpleSource`, call [`source::create_base_source_simple`]
+///     to create _base_ source stream of data and errors.
+///     - If the requested source is a `SourceReader`, call [`source::create_base_source`] to create
+///     a _base_ source stream (of various types, see [`SourceType`].
+///     - This is also where _reclocking_ happens.
+///   - Use a `render_decode_` function to decode the _base source_, if needed.
+///   - Render the [`SourceEnvelope`], which converts the stream of data from the upstream
+///   service into a _differential_ stream.
+///   - Perform various transformations, such as computing pre-declared linear operators and more.
+///
+/// There are other edge cases, like `render_decode_cdcv2` also implementing logic that would
+/// usually be part of [`SourceEnvelope`] rendering.
+///
+/// As you can see, while there are a large number of paths, the general, critical-path
+/// transformation is:
+/// - Create a _base_ source ([`source`]), which is a stream of data from upstream. (TODO(guswynn):
+/// consider renaming that module)
+/// - Decode that data ([`crate::decode`]).
+/// - Render the [`SourceEnvelope`], which converts that stream into a _differential_
+/// [`Collection`].
```

Comment:
This comment is great content, but it also a _huge_ bandaid IMO. This comment is essentially a line-by-line description of how this function works. I would much, much rather this _function_ be its own description of how it works. 

Rather than attempting to explain the complexity within, I contend that we should be refactoring this function into smaller functions, each with a descriptive name. ISTM that roughly each bullet point in your explanation should be its own function.

That way there is only _one_ description of how the code works—the code itself—and we don't have to worry about the two getting out of sync.